mpi-proxy-split/mpi_plugin.cpp

```diff
@@ -1122,6 +1165,36 @@ mpi_plugin_event_hook(DmtcpEvent_t event, DmtcpEventData_t *data)
   }
 }
 
+void
+global_p2p_communication_barrier()
+{
+  if (global_p2p_communication == 1 ||
+      (global_p2p_communication == 0 && internal_p2p_communication == 1))
+    return;
+
+  time_t my_time = time(NULL);
```

Comment:
If I understand this code, a call to MPI_Send will always call this routine, which will always print a message.  Computing a time and then doing a print is very expensive compared to MPI_Send.  So, we should compute a time (and do some printing) only in the uncommon case, and not in the common case.

Do I understand the code correctly?  If this is not a problem, then could you add some comments somewhere to make it clear that this happens only in the uncommon case?

Maybe my confusion comes from not understanding the variable `global_p2p_communication`.  I think it's a global variable initialized to 1, which would mean that we will not compute a time in the common case.  But then, can you add some comments there about `global_p2p_communication` and `internal_p2p_communication`?

Finally, should we be printing only in the case that we want to debug (`#ifdef DEBUG`)?  I'm not sure that all of the end users in the production code need to see this information.